iphone/Maps/Classes/CustomViews/MapViewControls/Welcome/PageController/MWMPageController.mm

```diff
@@ -109,12 +109,21 @@ - (void)setCurrentController:(MWMWelcomeController *)currentController
     return;
   self.isAnimatingTransition = YES;
   __weak auto weakSelf = self;
-  [self setViewControllers:@[ currentController ]
-                 direction:UIPageViewControllerNavigationDirectionForward
+  NSArray<UIViewController *> * viewControllers = @[ currentController ];
+  UIPageViewControllerNavigationDirection const direction = UIPageViewControllerNavigationDirectionForward;
+  [self setViewControllers:viewControllers
+                 direction:direction
                   animated:YES
                 completion:^(BOOL finished)
   {
-    weakSelf.isAnimatingTransition = NO;
+    if (finished)
+    {
+      dispatch_async(dispatch_get_main_queue(), ^
```

Comment:
Зачем нужен этот хак? Можно пояснить, пожалуйста? Это совершенно не очевидно.

> On Apr 8, 2016, at 15:52, Vlad Mihaylenko <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/CustomViews/MapViewControls/Welcome/PageController/MWMPageController.mm <https://github.com/mapsme/omim/pull/2768#discussion_r59019749>:
> 
> >                    animated:YES
> >                  completion:^(BOOL finished)
> >    {
> > -    weakSelf.isAnimatingTransition = NO;
> > +    if (finished)
> > +    {
> > +      dispatch_async(dispatch_get_main_queue(), ^
> это хак.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2768/files/382cc9159abb5ba03974776c2552b67e75e35065#r59019749>